tests/accelerators/test_ddp.py

```diff
@@ -83,3 +86,23 @@ def test_cli_to_pass(tmpdir, args=None):
     This test verify we can call function using test_cli name
     """
     return '1'
+
+
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
```

Comment:
The test should not require multi-gpu. You can mock `torch.cuda` to check since it will fail before any computation is done:

```python
@mock.patch.dict(os.environ, {"TORCH_DISTRIBUTED_BACKEND": "undefined"})
def test_torch_distributed_backend_env_variables(tmpdir, monkeypatch):
    """
    This test set `undefined` as torch backend and should raise an `Backend.UNDEFINED` ValueError.
    """
    monkeypatch.setattr("torch.cuda.device_count", lambda: 2)
    with pytest.raises(ValueError, match="Invalid backend: 'undefined'"):
        model = BoringModel()
        trainer = Trainer(
            default_root_dir=tmpdir,
            fast_dev_run=True,
            accelerator="ddp",
            gpus=2,
        )
        trainer.fit(model)
```

Reply:
you might also need to mock `torch.cuda.is_available` as well

Reply:
I tried it as it is and seemed to work. But we can also mock it ofc

Reply:
Done !